adapters/akka-http/src/test/scala/caliban/AkkaHttpAdapterSpec.scala

```diff
@@ -13,45 +14,44 @@ import zio._
 import zio.clock.Clock
 import zio.console.Console
 import zio.duration._
-import zio.internal.Platform
 import zio.random.Random
 import zio.test.{ DefaultRunnableSpec, TestFailure, ZSpec }
 
-import scala.concurrent.ExecutionContextExecutor
 import scala.language.postfixOps
 
 object AkkaHttpAdapterSpec extends DefaultRunnableSpec {
 
-  implicit val system: ActorSystem                                                            = ActorSystem()
-  implicit val executionContext: ExecutionContextExecutor                                     = system.dispatcher
-  implicit val runtime: Runtime[TestService with Console with Clock with Random with Uploads] =
-    Runtime.unsafeFromLayer(
-      TestService.make(sampleCharacters) ++ Console.live ++ Clock.live ++ Random.live ++ Uploads.empty,
-      Platform.default
-    )
-
-  val apiLayer: ZLayer[zio.ZEnv, Throwable, Has[Unit]] =
+  val apiLayer: ZLayer[zio.ZEnv, Throwable, TestService] =
     (for {
+      runtime     <- ZManaged.runtime[TestService with Console with Clock with Random with Uploads]
+      system      <- ZManaged.make(Task.effectTotal(ActorSystem()))(sys => ZIO.fromFuture(_ => sys.terminate()).ignore)
+      ec           = system.dispatcher
+      mat          = Materializer(system)
+      service     <- ZManaged.service[TestService.Service]
       interpreter <- TestApi.api.interpreter.toManaged_
       route        = path("api" / "graphql") {
-                       AkkaHttpAdapter.makeHttpService(interpreter, requestInterceptor = FakeAuthorizationInterceptor.bearer)
+                       AkkaHttpAdapter.makeHttpService(interpreter, requestInterceptor = FakeAuthorizationInterceptor.bearer)(
+                         runtime,
+                         implicitly,
+                         implicitly
+                       )
                      } ~ path("upload" / "graphql") {
-                       AkkaHttpAdapter.makeHttpUploadService(interpreter)
+                       AkkaHttpAdapter.makeHttpUploadService(interpreter)(runtime, implicitly, implicitly, implicitly)
                      } ~ path("ws" / "graphql") {
-                       AkkaHttpAdapter.makeWebSocketService(interpreter)
+                       AkkaHttpAdapter.makeWebSocketService(interpreter)(ec, runtime, mat, implicitly, implicitly)
                      }
-      _           <- ZIO
-                       .fromFuture(_ => Http().newServerAt("localhost", 8088).bind(route))
-                       .toManaged(server =>
-                         ZIO.fromFuture(_ => server.unbind()).ignore *> ZIO.fromFuture(_ => system.terminate()).ignore
-                       )
+      _           <- ZIO.fromFuture { _ =>
+                       implicit val s: ActorSystem = system
+                       Http()(system).newServerAt("localhost", 8088).bind(route)
```

Comment:
Do we need the implicit if we pass it explicitly?

Reply:
Nope, I'll remove